File: src/code/java/array/prefix_sum.java
public int[] fn(int[] arr) {
    int n = arr.length;
    int[] prefix = new int[n];
    prefix.add(arr[0]);

    for (int i = 1; i < n; i++) {
        prefix.add(prefix.get(i - 1) + arr[i]);
    }

    return prefix;
}

File: src/code/ruby/array/prefix_sum.rb
# This method computes the prefix sum of an array.
# It returns a new array where each element at index i is the sum of elements from index 0 to i-1 in the original array.
# # For example, given the array [1, 2, 3], the prefix sum array will be [1, 3, 6].
def prefix_sum(arr)
  n       = arr.length
  prefix  = [arr[0]]

  (1...n).each do |i|
    prefix << (prefix.last + arr[i])
  end

  prefix
end
